src/rate_limiter.cc

```diff
@@ -274,10 +350,31 @@ RateLimiter::DequeuePayload(
   (*payload)->Callback();
   if ((*payload)->GetInstance() == nullptr) {
     (*payload)->SetInstance(instances.front());
+    // Enqueue did not specify the specific instance to
+    // run with the payload. Hence, need to explicitly
+    // decrement the consumer count for the instance
+    // which got allocated.
+    payload_queue->specific_queues_[instances.front()]
+        ->DecrementConsumerCount();
```

Comment:
Lines 313 to 316 increment both the model scoped consumer count and the instance specific instance count. Shouldn't we have to decrement the model scoped consumer count here as well?

Reply:
The model scoped counter is decremented when we EnqueuePayload the request to the RateLimiter in L258. For accurate handling we should decrement the specific_queues_ count within EnqueuePayload call itself. However, we are not aware of the association between the instances being listened to by the consumer threads.